controllers/utils/base_reconciler.go

```diff
@@ -0,0 +1,408 @@
+	}
+	return reconcile.Result{}, nil
+}
+
+// GetReconcilationSubject gets the reconciliation subject of type defined by the concrete reconciler. It also sets up
+// a patch helper at this point.
+func (r *ReconciliationRunner) GetReconciliationSubject() (res ctrl.Result, reterr error) {
+	r.Log.V(1).Info("Getting reconciliation subject.")
+	err := client.IgnoreNotFound(r.Client.Get(r.RequestCtx, r.Request.NamespacedName, r.ReconciliationSubject))
+	if r.ReconciliationSubject.GetName() == "" {
+		r.SetReturnEarly()
+	}
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "error encountered while fetching reconciliation subject.")
+	}
+	return r.SetupPatcher()
+}
+
+// SetReconciliationSubjectToConcreteSubject sets reconciliation subject to passed concrete object.
+func (r *ReconciliationRunner) SetReconciliationSubjectToConcreteSubject(subject client.Object) CloudStackReconcilerMethod {
+	return func() (ctrl.Result, error) {
+		r.ReconciliationSubject = subject
+		return ctrl.Result{}, nil
+	}
+}
+
+// InitFromMgr just initiates a ReconcilerBase using given manager's fields/methods.
+func (r *ReconcilerBase) InitFromMgr(mgr ctrl.Manager, client cloud.Client) {
+	r.Client = mgr.GetClient()
+	r.BaseLogger = ctrl.Log.WithName("controllers").WithName("name")
+	r.Scheme = mgr.GetScheme()
+	r.CS = client
+}
+
+// GetParent returns the object owning the current resource of passed kind.
+func (r *ReconciliationRunner) GetParent(child client.Object, parent client.Object) CloudStackReconcilerMethod {
+	return func() (ctrl.Result, error) {
+		err := GetOwnerOfKind(r.RequestCtx, r.Client, child, parent)
+		return ctrl.Result{}, err
+	}
+}
+
+// GetOwnerOfKind uses the ReconciliationSubject's owner references to get the owner object of kind passed.
+func (r *ReconciliationRunner) GetOwnerOfKind(owner client.Object) CloudStackReconcilerMethod {
+	return func() (ctrl.Result, error) {
+		err := GetOwnerOfKind(r.RequestCtx, r.Client, r.ReconciliationSubject, owner)
+		return ctrl.Result{}, err
+	}
+}
+
+// NewChildObjectMeta creates a meta object with ownership reference and labels matching the current cluster.
+func (r *ReconciliationRunner) NewChildObjectMeta(name string) metav1.ObjectMeta {
+	ownerGVK := r.ReconciliationSubject.GetObjectKind().GroupVersionKind()
+	return metav1.ObjectMeta{
+		Name:      strings.ToLower(name),
+		Namespace: r.Request.Namespace,
+		Labels: map[string]string{capiv1.ClusterLabelName: r.CAPICluster.Name,
+			infrav1.CloudStackClusterLabelName: r.CSCluster.ClusterName},
+		OwnerReferences: []metav1.OwnerReference{
+			*metav1.NewControllerRef(r.ReconciliationSubject, ownerGVK),
+		},
+	}
+}
+
+// RequeueIfMissingBaseCRDs checks that the ReconciliationSubject, CAPI Cluster, and CloudStackCluster objects were
+// actually fetched and reques if not. The base reconciliation stages will continue even if not so as to allow deletion.
+func (r *ReconciliationRunner) RequeueIfMissingBaseCRDs() (ctrl.Result, error) {
```

Comment:
Naming nit: this is checking for CR's, not CRDs.

Reply:
Ah, definitely agree!

Reply:
bump since I don't think this was addressed